src/sqlfluff/rules/L003.py

```diff
@@ -328,27 +438,14 @@ def _eval(self, segment, raw_stack, memory, **kwargs):
             if this_line["indent_size"] > 0:
                 self.logger.debug("    Indented First Line. #%s", this_line_no)
                 return LintResult(
-                    anchor=segment,
+                    anchor=trigger_segment,
                     memory=memory,
                     description="First line has unexpected indent",
                     fixes=[
                         LintFix("delete", elem) for elem in this_line["indent_buffer"]
                     ],
                 )
 
-        # Are we linting a placeholder, and if so are we allowed to?
```

Comment:
As noted earlier, removing this undocumented, untested, apparently not working code.